includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -376,185 +376,221 @@ public function payment_fields() {
-				}
+	public function process_payment_for_order( $order, $cart, $payment_information, $manual_capture ) {
+		// phpcs:ignore WordPress.Security.NonceVerification.Missing
+		$save_payment_method = ! $payment_information->is_using_saved_card() && ( ! empty( $_POST[ 'wc-' . self::GATEWAY_ID . '-new-payment-method' ] ) || $payment_information->is_first_installment() );
+
+		$order_id = $order->get_id();
+		$amount   = $order->get_total();
+		$user     = $order->get_user() ?? wp_get_current_user();
+		$name     = sanitize_text_field( $order->get_billing_first_name() ) . ' ' . sanitize_text_field( $order->get_billing_last_name() );
+		$email    = sanitize_email( $order->get_billing_email() );
+		$metadata = [
+			'customer_name'  => $name,
+			'customer_email' => $email,
+			'site_url'       => esc_url( get_site_url() ),
+			'order_id'       => $order_id,
+		];
+		if ( $payment_information->is_first_installment() ) {
+			$metadata['payment_type'] = 'recurring';
+		}
 
-				// Update saved payment method information with checkout values, as some saved methods might not have billing details.
-				if ( $is_saved_method ) {
-					try {
-						$this->customer_service->update_payment_method_with_billing_details_from_order( $payment_method, $order );
-					} catch ( Exception $e ) {
-						// If updating the payment method fails, log the error message but catch the error to avoid crashing the checkout flow.
-						Logger::log( 'Error when updating saved payment method: ' . $e->getMessage() );
-					}
-				}
+		// Determine the customer making the payment, create one if we don't have one already.
+		$customer_id = $this->customer_service->get_customer_id_by_user_id( $user->ID );
 
-				$metadata = [
-					'customer_name'  => $name,
-					'customer_email' => $email,
-					'site_url'       => esc_url( get_site_url() ),
-					'order_id'       => $order->get_id(),
-				];
-
-				// Create intention, try to confirm it & capture the charge (if 3DS is not required).
-				$intent = $this->payments_api_client->create_and_confirm_intention(
-					WC_Payments_Utils::prepare_amount( $amount, 'USD' ),
-					'usd',
-					$payment_method,
-					$customer_id,
-					$manual_capture,
-					$save_payment_method,
-					$metadata,
-					$this->get_level3_data_from_order( $order )
-				);
+		if ( null === $customer_id ) {
+			// Create a new customer.
+			$customer_id = $this->customer_service->create_customer_for_user( $user, $name, $email );
+		} else {
+			// Update the existing customer with the current details. In the event the old customer can't be
+			// found a new one is created, so we update the customer ID here as well.
+			$customer_id = $this->customer_service->update_customer_for_user( $customer_id, $user, $name, $email );
+		}
 
-				$intent_id = $intent->get_id();
-				$status    = $intent->get_status();
+		// Update saved payment method information with checkout values, as some saved methods might not have billing details.
+		if ( $is_saved_method ) {
```

Comment:
PHP Notice:  Undefined variable: is_saved_method

Reply:
Thanks for catching this! I'm surprised I didn't see this on my end 🤔 

Fixed in 5502b59ab09495564b4bed6ed54fede6a53fb186!